sauna-app/components/map.tsx

```diff
@@ -0,0 +1,61 @@
+import 'leaflet/dist/leaflet.css';
+import { MapContainer,  TileLayer, Marker, Popup} from 'react-leaflet'
+import L from 'leaflet'
+import style from '../styles/Home.module.css'
+import { useState, useEffect } from 'react';
+import { SaunaData } from '../pages/typeDefinitions'
+
+
+
+function Map() {
+
+  const [saunas, setSaunas] = useState<SaunaData>();
+
+  useEffect(() => {
+    const fetchAPI = async () => {
+      await fetch('http://localhost:3000/api/saunas')
```

Comment:
A couple of things: You'll want to use `/api/saunas` instead of with the localhost, because it won't work once deployed with localhost. It's enough, because you're working inside Next. 

You also want to move fetching saunas to `index.tsx` and then just pass the saunas-array down to `Map`-component. You don't want to fetch any data in `components`-folder's components; only in the `pages`.